### PR TITLE
Update docs to remove iree folder that no longer exists

### PR DESCRIPTION
**Pull request by vwbaker**
I believe previously it was iree/... but it got refactored to be compiler/ and runtime/ folders so this command needs to be updated.

### CODE REVIEW

File: docs/developers/get_started/building_with_bazel_linux.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ $ python3 configure_bazel.py
 Run all core tests:
 
 ```shell
-$ bazel test -k iree/...
+$ bazel test -k compiler/... runtime/...
 ```
```
Comment (ScottTodd) [inline docs/developers/get_started/building_with_bazel_linux.md]: This script https://github.com/openxla/iree/blob/main/build_tools/bazel/build_core.sh is what our CI runs. That used to include directories, but the non-exhaustive list missed some paths. As of https://github.com/openxla/iree/pull/9206, you should just be able to `bazel test -k //...`

File: docs/developers/get_started/building_with_bazel_linux.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ $ python3 configure_bazel.py
 Run all core tests:
 
 ```shell
-$ bazel test -k iree/...
+$ bazel test -k compiler/... runtime/...
```
Comment (ScottTodd) [inline docs/developers/get_started/building_with_bazel_linux.md]: This time commenting with a suggestion (someone else was asking about this and I'd like to merge this)
```suggestion
$ bazel test -k //...
```

**Review by ScottTodd — COMMENTED** [review]
We should also update the other building_with_bazel_* docs

**Comment by vwbaker** [comment]
> We should also update the other building_with_bazel_* docs

Yeah true. Sorry I haven't gotten around to this yet; I was planning on updating once I actually got //... testing, but it doesn't work out of the box for me, and I haven't had the time to look into it yet. But this is probably still better than having //iree/... since that doesn't exist anymore so added.

File: docs/developers/get_started/building_with_bazel_linux.md (hunk 1 above)
Comment (vwbaker) [reply docs/developers/get_started/building_with_bazel_linux.md]: Yep, added. I just had to get my cuda environment set up correctly. This works at least to build now and the failing tests are probably mostly due to not having the right gpu

Comment (ScottTodd) [reply docs/developers/get_started/building_with_bazel_linux.md]: :+1: cool, this should be ready to merge once the other places are updated too: https://github.com/openxla/iree/tree/main/docs/developers/get_started (can just search the repo for `bazel test -k iree/...`)

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**